File: java/src/main/java/com/languageweaver/sdk/samples/translations/file/batch/CloudBatchFileTranslationService.java
package com.languageweaver.sdk.samples.translations.file.batch;

import com.languageweaver.sdk.common.cloud.CloudLanguageWeaverClient;
import com.languageweaver.sdk.common.constants.PdfConverter;
import com.languageweaver.sdk.translate.cloud.request.CloudTranslateBatchFileRequest;

import java.io.IOException;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ExecutionException;

public class CloudBatchFileTranslationService {

    public static void main(String[] args) throws Exception {
        try (CloudLanguageWeaverClient lwClient = new CloudLanguageWeaverClient().build()) {
            translateBatchFile(lwClient);
            translateBatchFileWithLinguisticOptions(lwClient);
        }
    }

    private static void translateBatchFile(CloudLanguageWeaverClient lwClient) throws ExecutionException, InterruptedException, IOException {
        CloudTranslateBatchFileRequest translateFileRequest = new CloudTranslateBatchFileRequest()
                .setSourceLanguageId("eng")
                .setTargetLanguageId("fra")
                .setModel("generic")
                .setPdfConverter(PdfConverter.ABBYY)
                .addDictionary("689f06cf-36ba-4903-a530-da1f7766f478")
                .addDictionary("3d297ee3-0878-4ef7-9ee7-ca14b48e6956")
                // provide full path to the input and output folders
                .setInputBatchFolderPath(Paths.get("java", "src", "main", "resources", "input").toFile().getAbsolutePath())
                .setOutputBatchFolderPath(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath());

        lwClient.translateBatchFile(translateFileRequest);
    }

    private static void translateBatchFileWithLinguisticOptions(CloudLanguageWeaverClient lwClient) throws ExecutionException, InterruptedException, IOException {
        Map<String, String> linguisticOptions = new HashMap<>();
        linguisticOptions.put("Spelling", "US");
        CloudTranslateBatchFileRequest translateFileRequest = new CloudTranslateBatchFileRequest()
                .setSourceLanguageId("fra")
                .setTargetLanguageId("eng")
                .setModel("generic")
                .setLinguisticOptions(linguisticOptions)
                .setPdfConverter(PdfConverter.ABBYY)
                .addDictionary("689f06cf-36ba-4903-a530-da1f7766f478")
                .addDictionary("3d297ee3-0878-4ef7-9ee7-ca14b48e6956")
                // provide full path to the input and output folders
                .setInputBatchFolderPath(Paths.get("java", "src", "main", "resources", "input").toFile().getAbsolutePath())
                .setOutputBatchFolderPath(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath());

        lwClient.translateBatchFile(translateFileRequest);
    }
}

File: javascript/browser/src/samples/translations/file/batch/CloudBatchFileTranslationService.js
import {
    ClientConfiguration,
    CloudLanguageWeaverClient,
    CloudTranslateBatchFileRequest,
    CredentialsConfiguration, PdfConverter
} from "@language-weaver/lw-sdk-js";
import {file1, file2, file3, file4} from "../../../../samples-utils";

export const translateBatchFileUsingCloud = async (clientId, clientSecret) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId, clientSecret);
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient()
            .withConfigurations(clientConfiguration)
            .build();
        const translateFileRequest = new CloudTranslateBatchFileRequest();
        translateFileRequest.sourceLanguageId = "eng";
        translateFileRequest.targetLanguageId = "fra";
        translateFileRequest.model = "generic";
        // provide list of dictionaries
        // translateFileRequest.addDictionary("689f06cf-36ba-4903-a530-da1f7766f478");
        // translateFileRequest.addDictionary("3d297ee3-0878-4ef7-9ee7-ca14b48e6956");
        // provide list of File objects
        translateFileRequest.input = [file1, file2];
        const translateFileResult = await cloudLanguageWeaverClient.translateBatchFileUsingCloudParams(translateFileRequest);
        console.log(translateFileResult.translationFileResults);
    } catch (e) {
        console.log(e);
    }
}

export const translatePdfBatchFileUsingCloud = async (clientId, clientSecret) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId, clientSecret);
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient()
            .withConfigurations(clientConfiguration)
            .build();
        const translateFileRequest = new CloudTranslateBatchFileRequest();
        translateFileRequest.sourceLanguageId = "eng";
        translateFileRequest.targetLanguageId = "fra";
        translateFileRequest.model = "generic";
        translateFileRequest.pdfConverter = PdfConverter.STANDARD;

        // provide list of File objects
        translateFileRequest.input = [file3, file4];
        const translateFileResult = await cloudLanguageWeaverClient.translateBatchFileUsingCloudParams(translateFileRequest);
        console.log(translateFileResult.translationFileResults);
    } catch (e) {
        console.log(e);
    }
}